File: DataStructureGo/queue.go
package main

type Queue struct {
	array []interface{}
}

func MakeQueue() Queue {
	return Queue{make([]interface{}, 0)}
}

func (q Queue) isEmpty() bool {
	if q.array == nil {
		return true
	}
	return len(q.array) == 0
}

func (q *Queue) enqueue(item interface{}) {
	if q.array == nil {
		q.array = make([]interface{}, 0)
	}
	q.array = append(q.array, item)
}

func (q *Queue) dequeue() interface{} {
	if q.isEmpty() {
		return nil
	}
	val := q.array[0]
	q.array = q.array[1:]
	return val
}

func (q Queue) peekFront() interface{} {
	if q.isEmpty() {
		return nil
	}
	return q.array[0]
}

File: DataStructureCSharp/DataStructure/Queue.cs
public class Queue<T> {

    private List<T> list { get; }

    public Queue() {
        list = new List<T>();
    }

    public void Enqueue(T item) {
        list.Add(item);
    }

    public T? Dequeue() {
        if (IsEmpty) {
            return default;
        }
        var first = list[0];
        list.RemoveAt(0);
        return first;
    }

    public T? Peek() {
        if (IsEmpty) {
            return default;
        }
        return list[0];
    }

    public bool IsEmpty {
        get {
            return list.Count == 0;
        }
    }

    public override string ToString() {
        return $"Queue: [{string.Join(", ", list)}]";
    }

}